packages/sprotty/src/features/viewport/zoom.ts

```diff
@@ -51,20 +51,32 @@ export class ZoomMouseListener extends MouseListener {
 
     override wheel(target: SModelElement, event: WheelEvent): Action[] {
         const viewport = findParentByFeature(target, isViewport);
-        if (viewport) {
-            const newZoom = this.getZoomFactor(event);
-            const viewportOffset = this.getViewportOffset(target.root, event);
-            const offsetFactor = 1.0 / (newZoom * viewport.zoom) - 1.0 / viewport.zoom;
-            const newViewport: Viewport = {
-                scroll: {
-                    x: viewport.scroll.x - offsetFactor * viewportOffset.x,
-                    y: viewport.scroll.y - offsetFactor * viewportOffset.y
-                },
-                zoom: viewport.zoom * newZoom
-            };
-            return [SetViewportAction.create(viewport.id, newViewport, { animate: false })];
+        if (!viewport) {
+            return [];
         }
-        return [];
+        const newViewport = () => {
```

Comment:
Why is this a function? It's executed immediately before the action is created.

Reply:
It's not a function, it's a `newViewport` supplier! :)
It looked ugly when using return in if/else branches.  
I will have a closer look soon, maybe extract some code to a protected  method and refactor existing code a bit

Reply:
Extracted zoom and scroll viewport calculation in separate functions for easier customization.